### ISSUE TITLE
Always open links in Incognito crashes

### ISSUE THREAD

**Issue by syphyr** [bromite/bromite#2482]
### Preliminary checklist

- [X] I have read the [README](https://github.com/bromite/bromite/blob/master/README.md).
- [X] I have searched the existing issues for my problem. This is a new ticket, NOT a duplicate or related to another open issue.
- [X] I have read the [FAQs](https://github.com/bromite/bromite/blob/master/FAQ.md).
- [X] I have updated Bromite to the latest version. The bug is reproducible on this latest version.
- [X] This is a bug report about the Bromite browser (not about the website, building Bromite, F-Droid or anything else).

### Can the bug be reproduced with corresponding Chromium version?

No

### Bromite version

108.0.5359.75

### Device architecture

arm64-v8a

### Android version

7.1

### Device model

T813

### Changed flags

no flags changed

### Is this bug about the SystemWebView?

No

### Is this bug happening in an incognito tab?

Yes

### Is this bug caused by the adblocker?

No

### Is this bug a crash?

--------- beginning of crash
12-03 03:06:16.586 15621 15621 F libc    : Fatal signal 5 (SIGTRAP), code 1 in tid 15621 (HTTP://WEBADDRESS.ELIDED)
12-03 03:06:17.211 15757 15757 I cr_ChildProcessService: Destroying ChildProcessService pid=15757
12-03 03:06:42.487 15880 15880 I cr_SplitCompatApp: Launched version=108.0.5359.75 minSdkVersion=23 isBundle=false processName=HTTP://WEBADDRESS.ELIDED isIsolated=false
12-03 03:06:42.662 15880 15880 E ResourceType: Style contains key with bad entry: 0x0101062c
12-03 03:06:42.684 15880 15880 E ResourceType: Style contains key with bad entry: 0x0101062c
12-03 03:06:42.738 15880 15905 I cr_LibraryLoader: Using LegacyLinker
12-03 03:06:42.740 15880 15905 W linker  : /data/app/HTTP://WEBADDRESS.ELIDED-1/lib/arm64/libchromium_android_linker.so: unused DT entry: type 0x70000001 arg 0x0
12-03 03:06:42.742 15880 15905 I cr_LibraryLoader: Loading chrome from within /data/app/HTTP://WEBADDRESS.ELIDED-1/HTTP://WEBADDRESS.ELIDED
12-03 03:06:42.751 15880 15880 I cr_CompositorSurfaceMgr: Transitioning to surface with format: -1
12-03 03:06:42.836 15880 15905 I cr_LibraryLoader: Successfully loaded native library
12-03 03:06:42.837 15880 15905 I cr_CachingUmaRecorder: Flushed 12 samples from 12 histograms.
12-03 03:06:42.838 15880 15897 I cr_tabmodel: Starting to fetch tab list for tab_state0
12-03 03:06:42.838 15880 15907 I cr_tabmodel: Starting to fetch tab list for tab_state1
12-03 03:06:42.839 15880 15907 I cr_tabmodel: State file does not exist.
12-03 03:06:42.840 15880 15897 I cr_tabmodel: Finished fetching tab list.
12-03 03:06:42.947 15880 15908 I Adreno  : QUALCOMM build                   : 422eead, I8e707a60c0
12-03 03:06:42.947 15880 15908 I Adreno  : Build Date                       : 11/23/17
12-03 03:06:42.947 15880 15908 I Adreno  : OpenGL ES Shader Compiler Version: XE031.09.00.04
12-03 03:06:42.947 15880 15908 I Adreno  : Local Branch                     : mybranch29262620
12-03 03:06:42.947 15880 15908 I Adreno  : Remote Branch                    : quic/LA.BR.1.3.6.c2_rb2.28
12-03 03:06:42.947 15880 15908 I Adreno  : Remote Branch                    : NONE
12-03 03:06:42.947 15880 15908 I Adreno  : Reconstruct Branch               : NOTHING
12-03 03:06:42.947 15880 15881 I art     : Do partial code cache collection, code=21KB, data=25KB
12-03 03:06:42.947 15880 15881 I art     : After code cache collection, code=21KB, data=25KB
12-03 03:06:42.947 15880 15881 I art     : Increasing code cache capacity to 128KB
12-03 03:06:42.954 15880 15908 I OpenGLRenderer: Initialized EGL, version 1.4
12-03 03:06:42.954 15880 15908 D OpenGLRenderer: Swap behavior 1
12-03 03:06:42.967 15880 15880 I cr_StartSurfaceConfig: Recorded HTTP://WEBADDRESS.ELIDED = 251 ms
12-03 03:06:43.065 15880 15880 W ThreadPoolForeg: type=1400 audit(0.0:6367): avc: denied { read } for name="iface_stat_fmt" dev="proc" ino=4026546301 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:proc_net:s0 tclass=file permissive=0
12-03 03:06:43.065 15880 15880 W ThreadPoolForeg: type=1400 audit(0.0:6368): avc: denied { read } for name="iface_stat_fmt" dev="proc" ino=4026546301 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:proc_net:s0 tclass=file permissive=0
12-03 03:06:43.079 15880 15937 W chromium: [WARNING:dns_client.cc(270)] BuildEffectiveConfig(): invalid configuration
12-03 03:06:43.086 15880 15937 W chromium: [WARNING:dns_client.cc(270)] BuildEffectiveConfig(): invalid configuration
12-03 03:06:43.126 15880 15930 I chromium: [INFO:verified_ruleset_dealer.cc(44)] OpenAndSetRulesetFile: /data/user/0/HTTP://WEBADDRESS.ELIDED Filter/Indexed Rules/35/1.2.3.4953/Ruleset Data is valid: 1
12-03 03:06:43.131 15880 15880 I cr_PlayServicesChecker: Disabling Background Sync because Play Services is not up to date.
12-03 03:06:43.151 15880 15930 I chromium: [INFO:indexed_ruleset.cc(41)] GetVerifyStatus: expected checksum = 0xa751c49, local checksum = 0xa751c49
12-03 03:06:43.175 15942 15942 I cr_SplitCompatApp: Launched version=108.0.5359.75 minSdkVersion=23 isBundle=false processName=HTTP://WEBADDRESS.ELIDED:privileged_process0 isIsolated=false
12-03 03:06:43.175 15880 15880 I cr_CompositorSurfaceMgr: Transitioning to surface with format: -1
12-03 03:06:43.180 15942 15942 I cr_ChildProcessService: Creating new ChildProcessService pid=15942
12-03 03:06:43.182 15880 15880 I cr_CompositorSurfaceMgr: surfaceCreated format: 0
12-03 03:06:43.193 15942 15956 I cr_LibraryLoader: Using LegacyLinker
12-03 03:06:43.196 15942 15962 W linker  : /data/app/HTTP://WEBADDRESS.ELIDED-1/lib/arm64/libchromium_android_linker.so: unused DT entry: type 0x70000001 arg 0x0
12-03 03:06:43.197 15942 15962 I cr_LibraryLoader: Loading chrome from within /data/app/HTTP://WEBADDRESS.ELIDED-1/HTTP://WEBADDRESS.ELIDED
12-03 03:06:43.221 15880 15880 I cr_PlayServicesChecker: Disabling Background Sync because Play Services is not up to date.
12-03 03:06:43.223 15880 15880 I cr_ChromeTabbedActivity: #initializeState
12-03 03:06:43.223 15880 15880 I cr_CipherFactory: #restoreFromBundle, no savedInstanceState.
12-03 03:06:43.223 15880 15880 I cr_tabmodel: #loadState, ignoreIncognitoFiles? true
12-03 03:06:43.224 15880 15880 I cr_tabmodel: Tab metadata, skipIncognitoCount? false incognitoCount: 1 totalCount: 1
12-03 03:06:43.226 15880 15880 I cr_tabmodel: #isIncognitoTabBeingRestored from tabDetails:  true
12-03 03:06:43.227 15880 15880 I cr_tabmodel: Failed to restore Incognito tab: its TabState could not be restored.
12-03 03:06:43.236 15880 15880 I cr_tabmodel: Loaded tab lists; counts: 0,0
12-03 03:06:43.237 15880 15880 I cr_ChromeTabbedActivity: #createInitialTab executed.
12-03 03:06:43.253 15880 15880 E ResourceType: Style contains key with bad entry: 0x0101062c
12-03 03:06:43.284 15942 15962 I cr_LibraryLoader: Successfully loaded native library
12-03 03:06:43.284 15942 15962 I cr_CachingUmaRecorder: Flushed 3 samples from 3 histograms.
12-03 03:06:43.296 15880 15880 E ResourcesCompat: Failed to find font-family tag
12-03 03:06:43.300 15942 15962 E libEGL  : validate_display:99 error 3008 (EGL_BAD_DISPLAY)
12-03 03:06:43.300 15942 15962 I Adreno  : QUALCOMM build                   : 422eead, I8e707a60c0
12-03 03:06:43.300 15942 15962 I Adreno  : Build Date                       : 11/23/17
12-03 03:06:43.300 15942 15962 I Adreno  : OpenGL ES Shader Compiler Version: XE031.09.00.04
12-03 03:06:43.300 15942 15962 I Adreno  : Local Branch                     : mybranch29262620
12-03 03:06:43.300 15942 15962 I Adreno  : Remote Branch                    : quic/LA.BR.1.3.6.c2_rb2.28
12-03 03:06:43.300 15942 15962 I Adreno  : Remote Branch                    : NONE
12-03 03:06:43.300 15942 15962 I Adreno  : Reconstruct Branch               : NOTHING
12-03 03:06:43.336 15880 15880 E cr_IntentUtils: getBooleanExtra failed on intent null
12-03 03:06:43.336 15880 15880 E cr_IntentUtils: getBooleanExtra failed on intent null
12-03 03:06:43.447 15880 15880 D skia    : ---- fAsset->read(8192) returned 0
12-03 03:06:43.447 15880 15880 D skia    : --- SkAndroidCodec::NewFromStream returned null
12-03 03:06:43.536 15880 15880 I cr_InactivityTracker: Last background time read from the SharedPreference is:-1.
12-03 03:06:43.573 15880 15880 I Choreographer: Skipped 34 frames!  The application may be doing too much work on its main thread.
12-03 03:06:43.603 15880 15881 I art     : Do partial code cache collection, code=57KB, data=61KB
12-03 03:06:43.603 15880 15881 I art     : After code cache collection, code=48KB, data=58KB
12-03 03:06:43.604 15880 15881 I art     : Increasing code cache capacity to 256KB
12-03 03:06:43.689 15880 15880 I cr_tabmodel: Appending tabs being restored to metadata lists, 0, startingNormalCount: 0, startingIncognitoCount: 1
12-03 03:06:43.689 15880 15880 I cr_tabmodel: Serializing tab lists; counts: 0, 1
12-03 03:06:43.698 15880 15880 W InputMethodManager: Ignoring onBind: cur seq=2135, given seq=2134
12-03 03:06:43.703 15880 15880 I cr_BromiteInlineUpdateController: BromiteUpdater: disabled by user
12-03 03:06:43.703 15880 15880 I cr_BromiteInlineUpdateController: BromiteUpdater: disabled by user
12-03 03:06:43.746 15880 15880 I chromium: [INFO:adblock_updater_service.cc(83)] AdBlockUpdaterService: starting up. First update attempt will take place in 5 seconds. Next update attempt will take place in 604800 seconds. 
12-03 03:06:43.757 15880 15909 I cr_BindingManager: Visible binding enabled: maxSize=40
12-03 03:06:43.764 15880 15928 E chromium: [ERROR:directory_reader_posix.cc(42)] opendir /data/user/0/HTTP://WEBADDRESS.ELIDED: No such file or directory (2)
12-03 03:06:43.779 15880 15939 I cr_LogcatCrashExtractor: Trying to extract logcat for minidump b2f434fe-9b9d-4a7f-ad5c-b13d8f6c816c.dmp15621.

### Describe the bug

Bromite crashes when any link is opened when "Always open links in Incognito" is enabled in settings.

### Steps to reproduce the bug

1. Enable "Always open links in Incognito" in settings
2. Open link

### Expected behavior

Does not crash

**Comment by kokukyocho** [comment]
Hello @syphyr!

I am a bot :robot:

Thanks for submitting this issue!
Since this bug is reproducible with Chromium please report it upstream on the [Chromium issue tracker](https://bugs.chromium.org/p/chromium/issues/list) and do not forget to read [Chromium project bug reporting guidelines](https://www.chromium.org/for-testers/bug-reporting-guidelines) first.

**Comment by vrsovin** [comment]
I experienced the same issue in last version (108.0.5359.75). I did test arm and arm64 version, both crash when I try to search directly in the app or try to open a link by external apps.

I am with android 12 and I did come back to previous version (107.0.5304.114).

**Comment by js290** [comment]
Yep, Settings -> Privacy and security -> Incognito mode -> Always open links in incognito on version 108.0.5359.75.
My [logcat.txt](https://github.com/bromite/bromite/files/10145650/logcat.txt).

**Comment by ghost** [comment]
I am facing the same issue.

**Bromite version**
108.0.5359.75

**Device architecture**
arm64-v8a

**Android version**
13

**Device model**
Pixel 4XL

**Changed flags**
None

[Logcat.md](https://github.com/bromite/bromite/files/10145957/Logcat.md)

**Comment by Disqu3-mirOir-qt** [comment]
same for me

**Comment by uazo** [comment]
@csagan5 to reply to https://github.com/bromite/bromite/issues/2419#issuecomment-1335679036

[c472ada1-7453-4e47-8565-60e7551bbb40.dmp11047.dump.zip](https://github.com/bromite/bromite/files/10145990/c472ada1-7453-4e47-8565-60e7551bbb40.dmp11047.dump.zip) my dump for this crash.

```
autoninja -C out/bromite minidump_stackwalk dump_syms
F=c472ada1-7453-4e47-8565-60e7551bbb40.dmp11047.dump && ~/chromium/src/out/bromite/minidump_stackwalk $F out/Symbols_arm64/ >~/$F.txt
```
where `out` is `https://github.com/bromite/bromite/releases/download/108.0.5359.75/arm64_ChromePublic_debug.7z` unpacked.

![image](https://user-images.githubusercontent.com/29201891/205432321-d3472a1b-3e48-4f61-8c3d-258fb4ec9435.png)

result [c472ada1-7453-4e47-8565-60e7551bbb40.dmp11047.dump.txt](https://github.com/bromite/bromite/files/10146001/c472ada1-7453-4e47-8565-60e7551bbb40.dmp11047.dump.txt)

cleaned:

```
 0  libchrome.so!partition_alloc::internal::PartitionRefCount::DoubleFreeOrCorruptionDetected(unsigned int) [partition_ref_count.h : 316 + 0x0]
 1  libchrome.so!allocator_shim::internal::PartitionFree(allocator_shim::AllocatorDispatch const*, void*, void*) [partition_ref_count.h : 179 + 0x0]
13  libchrome.so!blink::TextMetrics* cppgc::MakeGarbageCollectedTrait<blink::TextMetrics>::Call<blink::Font const&, blink::TextDirection&, blink::TextBaseline, blink::TextAlign, WTF::String const&>(cppgc::AllocationHandle&, blink::Font const&, blink::TextDirection&, blink::TextBaseline&&, blink::TextAlign&&, WTF::String const&) [allocation.h : 244 + 0x9]
16  libchrome.so!blink::TextMetrics* cppgc::MakeGarbageCollectedTrait<blink::TextMetrics>::Call<blink::Font const&, blink::TextDirection&, blink::TextBaseline, blink::TextAlign, WTF::String const&>(cppgc::AllocationHandle&, blink::Font const&, blink::TextDirection&, blink::TextBaseline&&, blink::TextAlign&&, WTF::String const&) [allocation.h : 244 + 0xb]
38  libchrome.so!web_contents_delegate_android::WebContentsDelegateAndroid::NavigationStateChanged(content::WebContents*, content::InvalidateTypes) [WebContentsDelegateAndroid_jni.h : 159 + 0xc]
39  libchrome.so!web_contents_delegate_android::WebContentsDelegateAndroid::NavigationStateChanged(content::WebContents*, content::InvalidateTypes) [WebContentsDelegateAndroid_jni.h : 151 + 0x0]
40  libchrome.so!content::WebContentsImpl::NotifyNavigationStateChanged(content::InvalidateTypes) [web_contents_impl.cc : 2235 + 0xc]
41  libchrome.so!content::WebContentsImpl::DidChangeVisibleSecurityState() [web_contents_impl.cc : 2698 + 0x4]
42  libchrome.so!content::NavigationControllerImpl::RendererDidNavigate(content::RenderFrameHostImpl*, content::mojom::DidCommitProvisionalLoadParams const&, content::LoadCommittedDetails*, bool, bool, bool, content::NavigationRequest*) [navigation_controller_impl.cc : 4136 + 0x8]
43  libchrome.so!content::NavigationControllerImpl::RendererDidNavigate(content::RenderFrameHostImpl*, content::mojom::DidCommitProvisionalLoadParams const&, content::LoadCommittedDetails*, bool, bool, bool, content::NavigationRequest*) [navigation_controller_impl.cc : 4122 + 0x0]
44  libchrome.so!IPC::(anonymous namespace)::ChannelAssociatedGroupController::SendMessage(mojo::Message*) [lock.h : 34 + 0x4]
45  libchrome.so!operator new(unsigned long) [allocator_shim.cc : 159 + 0xc]
46  libchrome.so!mojo::internal::SendMojoMessage(mojo::MessageReceiver&, mojo::Message&) [send_message_helper.cc : 23 + 0x0]
47  libchrome.so!operator new(unsigned long) [allocator_shim.cc : 159 + 0xc]
48  libchrome.so!operator new(unsigned long) [allocator_shim.cc : 159 + 0xc]
49  libchrome.so!operator new(unsigned long) [allocator_shim.cc : 159 + 0xc]
50  libchrome.so!content::RenderFrameHostManager::GetProxyToOuterDelegate() [render_frame_host_manager.cc : 473 + 0x0]
```

then, that it means little in this particular case is just bad luck!

**Comment by csagan5** [comment]
> 12-03 03:06:43.336 15880 15880 E cr_IntentUtils: getBooleanExtra failed on intent null
> 12-03 03:06:43.336 15880 15880 E cr_IntentUtils: getBooleanExtra failed on intent null

@syphyr your crash log does not contain the stacktrace


> to reply to [#2419 (comment)](https://github.com/bromite/bromite/issues/2419#issuecomment-1335679036)
> 
> [c472ada1-7453-4e47-8565-60e7551bbb40.dmp11047.dump.zip](https://github.com/bromite/bromite/files/10145990/c472ada1-7453-4e47-8565-60e7551bbb40.dmp11047.dump.zip) my dump for this crash.
> 
> ```
> autoninja -C out/bromite minidump_stackwalk dump_syms
> F=c472ada1-7453-4e47-8565-60e7551bbb40.dmp11047.dump && ~/chromium/src/out/bromite/minidump_stackwalk $F out/Symbols_arm64/ >~/$F.txt
> ```
> 
> where `out` is `https://github.com/bromite/bromite/releases/download/108.0.5359.75/arm64_ChromePublic_debug.7z` unpacked.

Ok, thanks; I follow the same but I often do not get symbols and paths.

Each of these crashes (except the one on this issue, which contains no stacktrace) point to a different code location, but I can reproduce it so it should be easy to solve.

**Comment by uazo** [comment]
> but I can reproduce it so it should be easy to solve.

try to add https://github.com/uazo/bromite/blob/next-v108/build/patches/Add-an-always-incognito-mode.patch#L1775 you miss that one

**Comment by csagan5** [comment]
> > but I can reproduce it so it should be easy to solve.
> 
> try to add https://github.com/uazo/bromite/blob/next-v108/build/patches/Add-an-always-incognito-mode.patch#L1775 you miss that one

I did not add it on purpose; I will be trying a different patch.

**Comment by csagan5** [comment]
@uazo this will be my fix:
```diff
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java
@@ -210,7 +210,7 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
      */
     public void initializeWithNative() {
         mOptimizationsEnabled =
-                ChromeFeatureList.isEnabled(ChromeFeatureList.ANDROID_SCROLL_OPTIMIZATIONS);
+                ChromeFeatureList.isEnabled(ChromeFeatureList.ANDROID_SCROLL_OPTIMIZATIONS) && !mIsAlwaysIncognito;
         mLastUsedNonOTRProfile = Profile.getLastUsedRegularProfile();
         mNativeLocationBarModelAndroid = LocationBarModelJni.get().init(LocationBarModel.this);
 
```

And with no change on the destructor path; it works and the downside is that there are no optimisations in always incognito.

**Comment by uazo** [comment]
absolutely fine. I had thought about that as well, however, the downside is that if they tie other optimizations to that flag, in the future, we lose them. But it's all the same for now.

**Comment by csagan5** [comment]
Fixed in `108.0.5359.106`.
